src/sp/protocol/mqtt/mqtt_parser.c

```diff
@@ -583,16 +583,16 @@ conn_handler(uint8_t *packet, conn_param *cparam, size_t max)
 	NNI_GET16(packet + pos, tmp);
 	cparam->keepalive_mqtt = tmp;
 	pos += 2;
-	// properties
 
+	// properties
 	if (cparam->pro_ver == MQTT_PROTOCOL_VERSION_v5) {
 		// check length
 		log_trace("Decoding MQTT V5 Properties");
-		if (pos >= max)
+		if (pos + 4 >= max)
```

Comment:
4 == 2 (property len) + 2 (clientid len). Am I right?

Reply:
it's the req of `get_var_integer`.